src/Rules/ArrayOfRule.php

```diff
@@ -90,7 +90,7 @@ public function processValue($value, Args $args, FieldContext $context): array
 		if ($args->maxItems !== null && count($value) > $args->maxItems) {
 			$type->markParameterInvalid(self::MAX_ITEMS);
 
-			throw ValueDoesNotMatch::create($type);
+			throw ValueDoesNotMatch::create($type, $value);
```

Comment:
We should find a way how to reduce that value, we want to drop values above limit as soon as the processing is finished to save memory. It does not really matter anyway what the specific values were, just the fact they are above limit.

Reply:
Using reduced reference to value may be helpful anyway. e.g. for string with max length 100 is enough to have like 10 characters string with information that given string had 10k characters, no way we would dump it whole in error formatter

Reply:
We should pass whole value to stay consistent.
Other services may manipulate it as they need.

Or maybe add an extra arg which would tell if values above limit should be dropped?
But then we would have to slice that array and that's not for free either :/

Reply:
Don't solve it now, it's more a note for me :) I just want to be sure we don't create some memory bottleneck